### PR TITLE
Create book request

### PR DESCRIPTION
**Pull request by banguyenvareal**

### CODE REVIEW

File: app/assets/javascripts/book_requests.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (dinhvareal) [inline app/assets/javascripts/book_requests.coffee]: remove this file

File: app/assets/stylesheets/book_requests.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the BookRequests controller here.
```
Comment (dinhvareal) [inline app/assets/stylesheets/book_requests.scss]: delete this file

File: app/helpers/book_requests_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module BookRequestsHelper
```
Comment (dinhvareal) [inline app/helpers/book_requests_helper.rb]: Delete

File: app/models/book_request.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class BookRequest < ApplicationRecord
+  belongs_to :user
+  has_many :book_request_image, dependent: :destroy
+  validates :name, presence: true
+  validates :quantity, presence: true,
+                       numericality:
+                       { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :budget, presence: true,
+                     numericality: { only_integer: true, greater_than: 0 }
```
Comment (dinhvareal) [inline app/models/book_request.rb]: greater_than_or_equal_to

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>Bookstore</title>
+    <title><%= title_page(yield(:title)) %></title>
```
Comment (dinhvareal) [inline app/views/layouts/application.html.erb]: <title><%= (yield(:title)) %></title>

File: db/migrate/20190724062800_add_book_request_to_image.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddBookRequestToImage < ActiveRecord::Migration[5.2]
```
Comment (dinhvareal) [inline db/migrate/20190724062800_add_book_request_to_image.rb]: what is it for?

File: app/assets/javascripts/book_requests.coffee (hunk 1 above)
Comment (banguyenvareal) [reply app/assets/javascripts/book_requests.coffee]: removed

File: app/assets/stylesheets/book_requests.scss (hunk 2 above)
Comment (banguyenvareal) [reply app/assets/stylesheets/book_requests.scss]: deleted

File: app/helpers/book_requests_helper.rb (hunk 3 above)
Comment (banguyenvareal) [reply app/helpers/book_requests_helper.rb]: deleted

File: app/models/book_request.rb (hunk 4 above)
Comment (banguyenvareal) [reply app/models/book_request.rb]: fixed

File: app/views/layouts/application.html.erb (hunk 5 above)
Comment (banguyenvareal) [reply app/views/layouts/application.html.erb]: `fixed`

File: db/migrate/20190724062800_add_book_request_to_image.rb (hunk 6 above)
Comment (banguyenvareal) [reply db/migrate/20190724062800_add_book_request_to_image.rb]: removed

File: config/locales/en.yml [hunk 7]
```diff
@@ -108,3 +108,9 @@ en:
   shared:
     error_messages:
       errors_notification: "Have %{errors_count} errors in the form"
+  book_request:
+    create: "Create book request"
+    name: "Book's title:"
+    budget: "budget:"
```
Comment (mrcuong84) [inline config/locales/en.yml]: should be Capital

File: spec/models/book_request_spec.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  it { should validate_presence_of(:budget) }
+  it { should validate_numericality_of(:quantity).only_integer }
+  it do
+    should validate_numericality_of(:quantity).is_greater_than_or_equal_to(0)
+  end
+  it { should validate_numericality_of(:budget).only_integer }
+  it do
+    should validate_numericality_of(:budget).is_greater_than_or_equal_to(0)
+  end
+end
```
Comment (mrcuong84) [inline spec/models/book_request_spec.rb:17]: book request should belong to user

**Review by mrcuong84 — APPROVED** [review]

**Review by dinhvareal — APPROVED** [review]

File: app/models/book_request.rb [hunk 9]
```diff
@@ -0,0 +1,11 @@
+class BookRequest < ApplicationRecord
+  belongs_to :user
+  has_many :book_request_image, dependent: :destroy
```
Comment (hieunguyentrung) [inline app/models/book_request.rb]: `book_request_image` and write the test for it.

File: app/models/book_request.rb [hunk 10]
```diff
@@ -0,0 +1,11 @@
+class BookRequest < ApplicationRecord
+  belongs_to :user
```
Comment (hieunguyentrung) [inline app/models/book_request.rb:2]: Write test.

**Review by hieunguyentrung — CHANGES_REQUESTED** [review]

File: db/migrate/20190724062048_create_book_requests.rb [hunk 11]
```diff
@@ -0,0 +1,13 @@
+class CreateBookRequests < ActiveRecord::Migration[5.2]
+  def change
+    create_table :book_requests do |t|
+      t.string :name, null: false
+      t.text :comment
+      t.integer :budget, null: false
```
Comment (hieunguyentrung) [inline db/migrate/20190724062048_create_book_requests.rb]: Default value ?

**Review by hieunguyentrung — CHANGES_REQUESTED** [review]

File: db/migrate/20190724062048_create_book_requests.rb (hunk 11 above)
Comment (banguyenvareal) [reply db/migrate/20190724062048_create_book_requests.rb]: fixed

File: app/models/book_request.rb (hunk 10 above)
Comment (banguyenvareal) [reply app/models/book_request.rb:2]: Ready in `spec/models/book_request_spec.rb`

File: app/models/book_request.rb (hunk 9 above)
Comment (hieunguyentrung) [inline app/models/book_request.rb]: `book_request_images`

**Review by hieunguyentrung — CHANGES_REQUESTED** [review]

File: app/models/book_request.rb (hunk 9 above)
Comment (banguyenvareal) [reply app/models/book_request.rb]: fixed

**Review by hieunguyentrung — APPROVED** [review]

**Pull request merged**